internal/lifecycle/readme.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lifecycle
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/GoogleCloudPlatform/serverless-sample-tester/internal/util"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+var (
+	// The tag that should appear immediately before code blocks in a README to indicate that the enclosed commands
+	// are to be used by this program for building and deploying the sample.
+	codeTag = "{sst-run-unix}"
+
+	gcloudCommandRegexp   = regexp.MustCompile(`\bgcloud\b`)
+	cloudRunCommandRegexp = regexp.MustCompile(`\brun\b`)
+
+	gcrURLRegexp = regexp.MustCompile(`gcr.io/.+/\S+`)
+)
+
+// parseREADME parses a README file with the given name. It reads terminal commands surrounded by one of the codeTags
+// listed above and loads them into a Lifecycle. In the process, it replaces the Cloud Run service name and Container
+// Registry tag with the provided inputs.
```

Comment:
General comment looking at this function: parsing logic is pretty complicated and can be very tricky to get right in all the edge cases. How well-defined is the README format? Is doing this parsing a hard requirement for your project? If possible it would simplify things a lot if you could force users to provide input data as a json and then rely on the nice built in support that golang has for parsing those.

Reply:
Per offline discussion, we've decided that we'll keep pursuing it, but make sure to spend more time thinking about potential edge cases and keep an eye out for any that may come up. Also, will open an issue after this PR is merged for tests that will help us also help us catch edge cases.